Source/GSOrderedSet.m

```diff
@@ -0,0 +1,399 @@
+    
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+    
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free
+    Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02111 USA.
+*/
+
+#import "common.h"
+#import "Foundation/NSOrderedSet.h"
+#import "Foundation/NSAutoreleasePool.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSEnumerator.h"
+#import "Foundation/NSException.h"
+#import "Foundation/NSPortCoder.h"
+#import "Foundation/NSIndexSet.h"
+#import "Foundation/NSKeyedArchiver.h"
+#import "GNUstepBase/GSObjCRuntime.h"
+#import "GSPrivate.h"
+#import "GSFastEnumeration.h"
+#import "GSDispatch.h"
+#import "GSSorting.h"
+
+#define	GSI_ARRAY_TYPE	NSRange
+#define	GSI_ARRAY_NO_RELEASE	1
+#define	GSI_ARRAY_NO_RETAIN	1
+#define GSI_ARRAY_TYPES       GSUNION_OBJ
+
+#define GSI_ARRAY_RELEASE(A, X)	[(X).obj release]
+#define GSI_ARRAY_RETAIN(A, X)	[(X).obj retain]
+
+#import "GNUstepBase/GSIArray.h"
+
+//  static SEL     memberSel;
+static SEL      privateCountOfSel;
+@interface GSOrderedSet : NSOrderedSet
+{
+@public
+  GSIArray_t array;
+}
+@end
+
+@interface GSMutableOrderedSet : NSMutableOrderedSet
+{
+@public
+  GSIArray_t array;
+@private
+  NSUInteger _version;
+}
+@end
+
+@interface GSOrderedSetEnumerator : NSEnumerator
+{
+  GSOrderedSet *set;
+  unsigned      current;
+  unsigned      count;
+}
+@end
+
+@interface GSOrderedSetEnumeratorReverse : GSOrderedSetEnumerator
+@end
+
+@implementation GSOrderedSetEnumerator
+- (id) initWithOrderedSet: (NSOrderedSet*)d
+{
+  self = [super init];
+  if (self != nil)
+    {
+      set = (GSOrderedSet*)RETAIN(d);
+      current = 0;
+      count = GSIArrayCount(&set->array);
+    }
+  return self;
+}
+
+- (id) nextObject
+{
+  if(current < count)
+    {
+      GSIArrayItem item = GSIArrayItemAtIndex(&set->array, current);
+      current++;
+      return (id)(item.obj);
+    }
+  return nil;
+}
+
+- (void) dealloc
+{
+  RELEASE(set);
+  [super dealloc];
+}
+@end
+
+
+@implementation GSOrderedSetEnumeratorReverse
+- (id) initWithOrderedSet: (GSOrderedSet*)d
+{
+  self = [super initWithOrderedSet: d];
+  if(self != nil)
+    {
+      current = GSIArrayCount(&set->array);
+    }
+  return self;
+}
+
+- (id) nextObject
+{
+  GSIArrayItem item;
+
+  if (current == 0)
+    return nil;
+  
+  item = GSIArrayItemAtIndex(&set->array, --current);
+  return (id)(item.obj);
+}
+@end
+
+@implementation GSOrderedSet
+
+static Class	setClass;
+static Class	mutableSetClass;
+
++ (void) initialize
+{
+  if (self == [GSOrderedSet class])
+    {
+      setClass = [GSOrderedSet class];
+      mutableSetClass = [GSMutableOrderedSet class];
+      privateCountOfSel = @selector(_countForObject:);
```

Comment:
I think this no longer gets used anywhere. You may now even drop the whole method _countForObject:.